dotnet/src/SemanticKernel.Abstractions/AI/FunctionChoiceBehaviors/NoneFunctionChoiceBehavior.cs

```diff
@@ -0,0 +1,104 @@
+/// </remarks>
+[Experimental("SKEXP0001")]
+public sealed class NoneFunctionChoiceBehavior : FunctionChoiceBehavior
+{
+    /// <summary>
+    /// List of the functions that the model can choose from.
+    /// </summary>
+    private readonly IEnumerable<KernelFunction>? _functions;
+
+    /// <summary>
+    /// This class type discriminator used for polymorphic deserialization of the type specified in JSON and YAML prompts.
+    /// </summary>
+    public const string TypeDiscriminator = "none";
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="NoneFunctionChoiceBehavior"/> class.
+    /// </summary>
+    [JsonConstructor]
+    public NoneFunctionChoiceBehavior()
+    {
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="NoneFunctionChoiceBehavior"/> class.
+    /// </summary>
+    /// <param name="functions">The subset of the <see cref="Kernel"/>'s plugins' functions information.</param>
+    public NoneFunctionChoiceBehavior(IEnumerable<KernelFunction> functions)
+    {
+        this._functions = functions;
+        this.Functions = functions.Select(f => FunctionName.ToFullyQualifiedName(f.Name, f.PluginName)).ToList();
+    }
+
+    /// <summary>
+    /// Fully qualified names of subset of the <see cref="Kernel"/>'s plugins' functions information to provide to the model.
+    /// </summary>
+    [JsonPropertyName("functions")]
+    [JsonConverter(typeof(FunctionNameFormatJsonConverter))]
+    public IList<string>? Functions { get; set; }
+
+    /// <inheritdoc/>
+    public override FunctionChoiceBehaviorConfiguration GetConfiguration(FunctionChoiceBehaviorContext context)
+    {
+        List<KernelFunction>? availableFunctions = null;
+
+        // Handle functions provided via the 'Functions' property as function fully qualified names.
+        if (this.Functions is { } functionFQNs && functionFQNs.Any())
+        {
+            availableFunctions = [];
+
+            foreach (var functionFQN in functionFQNs)
+            {
+                var nameParts = FunctionName.Parse(functionFQN);
+
+                // Check if the function is available in the kernel.
+                if (context.Kernel!.Plugins.TryGetFunction(nameParts.PluginName, nameParts.Name, out var function))
+                {
+                    availableFunctions.Add(function);
+                    continue;
+                }
+
+                // Check if a function instance that was not imported into the kernel was provided through the constructor.
+                function = this._functions?.FirstOrDefault(f => f.Name == nameParts.Name && f.PluginName == nameParts.PluginName);
+                if (function is not null)
+                {
+                    availableFunctions.Add(function);
+                    continue;
+                }
+
+                throw new KernelException($"The specified function {functionFQN} is not available.");
```

Comment:
For None, does it matter? Nothing needs to be invoked.

Reply:
Yes, right, nothing will be invoked. However, if a subset of functions is specified, the caller wants them to be advertised to the model. If any is not found, it means it does not exist because it's either not registered in the kernel or the function name is incorrect. In either case, it seems right to inform the caller about it, doesn't it?